libs/agno/agno/models/message.py

```diff
@@ -257,6 +257,12 @@ def from_dict(cls, data: Dict[str, Any]) -> "Message":
                 else:
                     data["video_output"] = Video(**vid_data)
 
+        # Handle metrics deserialization - convert dict to MessageMetrics
+        if "metrics" in data and data["metrics"] is not None:
+            if isinstance(data["metrics"], dict):
```

Comment:
I hate these checks 😅 , it should never be the case that we are already deserialized